smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -299,36 +299,54 @@ private boolean writeRequestQueryString(
             HttpTrait trait
     ) {
         TypeScriptWriter writer = context.getWriter();
-        SymbolProvider symbolProvider = context.getSymbolProvider();
         List<HttpBinding> queryBindings = bindingIndex.getRequestBindings(operation, Location.QUERY);
+        List<HttpBinding> queryParamsBindings = bindingIndex.getRequestBindings(operation, Location.QUERY_PARAMS);
 
         // Build the initial query bag.
         Map<String, String> queryLiterals = trait.getUri().getQueryLiterals();
-        if (!queryLiterals.isEmpty() || !queryBindings.isEmpty()) {
+        if (!queryLiterals.isEmpty() || !queryBindings.isEmpty() || !queryParamsBindings.isEmpty()) {
             writer.openBlock("const query: any = {", "};", () -> {
                 if (!queryLiterals.isEmpty()) {
                     // Write any query literals present in the uri.
                     queryLiterals.forEach((k, v) -> writer.write("$S: $S,", k, v));
                 }
+                // Handle any additional query params bindings.
```

Comment:
nit: probably worth noting that these have to come first because explicit bindings override them and js will take whatever comes last

Reply:
Added in [`95f2120` (#311)](https://github.com/awslabs/smithy-typescript/pull/311/commits/95f21204f80c6795f9e21391b31fc102fd4cd972)